debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/OracleContainer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.testing.testcontainers;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.Future;
+
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+public class OracleContainer extends org.testcontainers.containers.OracleContainer {
+
+    private static final String FALLBACK_ORACLE_SERVER_VERSION = "21.3.0";
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("quay.io/rh_integration/dbz-oracle");
+    static final String DEFAULT_TAG = System.getProperty("version.oracle.server", FALLBACK_ORACLE_SERVER_VERSION);
+    private static final int ORACLE_DEFAULT_STARTUP_TIMEOUT_SECONDS = 240;
+    private static final int DEFAULT_CONNECT_TIMEOUT_SECONDS = 120;
+    static final int ORACLE_PORT = 1521;
+    private static final int APEX_HTTP_PORT = 8080;
+
+    public OracleContainer() {
+        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG).asCompatibleSubstituteFor("gvenzl/oracle-xe"));
+    }
+
+    public OracleContainer(String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName).asCompatibleSubstituteFor("gvenzl/oracle-xe"));
+    }
+
+    public OracleContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        preconfigure();
+    }
+
+    public OracleContainer(Future<String> dockerImageName) {
+        super(dockerImageName);
+        preconfigure();
+    }
+
+    private void preconfigure() {
+        withUsername("c##dbzuser").withPassword("dbz").withDatabaseName("ORCLCDB");
```

Comment:
We are specifying the `version.oracle.server` so that we can technically run the container with any flavor of Oracle. This means we could technically point this against an instance that doesn't support multitenancy.  In this case, the user `c##dbzuser` wouldn't be allowed since common-user designators are not permitted when there is no multi-tenancy.  I would perhaps recommend we source these from system properties much like the `DEFAULT_TAG` above so that this can be easily toggled.  wdyt?

Reply:
This also may mean that any test (which I think you have one) that explicitly looks for `ORCLPDB1` as the response would fail if we did point to non-CDB images as the only database there is the `ORCLCDB` one fwiw.

Reply:
I was looking at the `pom.xml` of the Oracle connector it looks like we are only passing the version number like `<version.oracle.server>19.3.0</version.oracle.server>` there and it looked like that only the version is speficied there, and nowhere it was including the xs or noncdb "flavors"? The `xtreams` (and oracle-all) profiles don't seem to define another image using this system property, nor I found it in the release config (dir jenkins-jobs) or the debezium-testing-system module. Do you have a pointer to where we set it to something else? Is it only used for development? Maybe we can split this into the version number (that I expect alone in that property) and an optional "flavor" extension for other profiles?

A related question: Should very different images like the noncdb or xs images be just another tag on the same image, or should those be in different repos? Like: `rh_integration/dbz-oracle`, `rh_integration/dbz-oracle-xs`, `rh_integration/dbz-oracle-noncdb`? That would avoid such confusion. When those flavors are not compatible they should maybe not be different tags in the same repo but in separate repos?

Reply:
So by default, `version.oracle.server` points to the `19.3.0` image, which is the LogMiner multi-tenancy image; however, one could use `-Dversion.oracle.server=19.3.0-noncdb` and use the LogMiner image that does not include multitenancy.  In this case, the `c##dbzuser` won't be valid because users cannot be prefixed with `c##` in those builds; that's only permissible with multitenancy.

This specifically has to do with how the database is configured.  The `19.3.0` image with CDB enabled is essentially a single instance that contains a ROOT database and one pluggable database (PDB), aka a tenant.  Without the CDB feature, the single instance is the only database and there is no concept of common vs local users.

The POM profiles don't get into the specifics around multi-tenancy because the only difference is the credentials and connector configuration used, which can all be controlled using `-D` command line arguments.

Where you'll find some reference is in Oracle's `TestHelper#getDatabaseConfig(String)` method.  We basically consume all systems properties that start with `database.` and those allow the CLI to override the user name, password, database name, and pluggable database names.

> Should very different images like the noncdb or xs images be just another tag on the same image, or should those be in different repos? Like: rh_integration/dbz-oracle, rh_integration/dbz-oracle-xs, rh_integration/dbz-oracle-noncdb?

It's simply a matter of preference.  It's no different than MySQL having  `8.0.34`, `8.0.34-debian`, etc.  We could certainly think about changing this if its a problem, but it's certainly not something that I'd be comfortable doing right now, given how much reliance some of our tooling is now on those specific tags.

> If you think that this is not valid or you want to keep the current structure, sure, I can use system properties for configuration!

So all I am suggesting is that rather than these be strictly hardcoded, we allow the following to override the hardcoded values from the CLI if they're provided:

* `database.user` for the username
* `database.password` for the password
* `database.dbname` for the database name (references where you are using `ORCLCDB`)
* `database.pdb.name` for the pluggable database (references where you may be using `ORCLPDB1`).

Reply:
Done (except `database.pdb.name`). I need some more guidance what exactly should change in the Oracle REST Extension tests. I filed https://issues.redhat.com/browse/DBZ-7091 for this.
Could you merge the PR when you are back?